app/Providers/AppServiceProvider.php

```diff
@@ -44,6 +46,12 @@ public function boot()
      */
     public function register()
     {
-        //
+        $this->app->singleton(ContactRepository::class, function(){
```

Comment:
This looks complex, but it's really not. At it's core we have the `ContactRepositoryEloquent`, this uses eloquent to store and fetch contacts.

Then we have the `ContactRepositoryCacheFilesystem`, this is a cache implementation that uses the file system. It's a decorator around the `ContactRepository`. 

Finally we have the `ContactRepositoryTimer`, this is also a decorator, and it simply time show long it took to fetch the list.

Since we're using decorators here, it's very easy to change the cache implementation, or to remove the timer.